src/mscorlib/src/System/IO/FileStream.cs

```diff
@@ -1163,9 +1163,11 @@ private unsafe void VerifyHandleIsSync()
             set {
                 if (value < 0) throw new ArgumentOutOfRangeException("value", Environment.GetResourceString("ArgumentOutOfRange_NeedNonNegNum"));
                 Contract.EndContractBlock();
-                if (_writePos > 0) FlushWrite(false);
-                _readPos = 0;
-                _readLen = 0;
+                if (_writePos > 0) {
+                    FlushWrite(false);
+                    _readPos = 0;
+                    _readLen = 0;
+                }
```

Comment:
What is this fixing?  What tests are failing without this change?

Also, this FileStream is only used internally by mscorlib; the one used publically is in the corefx repo.